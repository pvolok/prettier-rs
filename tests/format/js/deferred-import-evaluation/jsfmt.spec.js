const importDeferTests = [
  "import-defer.js",
  "import-defer-attributes-declaration.js",
];
const invalidSyntaxTests = ["no-default.js", "no-named.js"];

run_spec(import.meta, ["babel"], {
  errors: {
    acorn: [...importDeferTests, ...invalidSyntaxTests],
    espree: [...importDeferTests, ...invalidSyntaxTests],
    meriyah: [...importDeferTests, ...invalidSyntaxTests],
    babel: invalidSyntaxTests,
    __babel_estree: invalidSyntaxTests,
  },
});
