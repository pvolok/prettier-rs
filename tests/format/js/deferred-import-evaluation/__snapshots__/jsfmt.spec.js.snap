// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-defer.js [acorn] format 1`] = `
"Unexpected token (1:14)
> 1 | import defer * as ns from "x";
    |              ^
  2 |"
`;

exports[`import-defer.js [espree] format 1`] = `
"Unexpected token * (1:14)
> 1 | import defer * as ns from "x";
    |              ^
  2 |"
`;

exports[`import-defer.js [meriyah] format 1`] = `
"Import source must be string (1:14)
> 1 | import defer * as ns from "x";
    |              ^
  2 |"
`;

exports[`import-defer.js format 1`] = `
====================================options=====================================
parsers: ["babel"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import defer * as ns from "x";

=====================================output=====================================
import defer * as ns from "x";

================================================================================
`;

exports[`import-defer-attributes-declaration.js [acorn] format 1`] = `
"Unexpected token (1:14)
> 1 | import defer * as ns from "x" with { attr: "val" };
    |              ^
  2 |"
`;

exports[`import-defer-attributes-declaration.js [espree] format 1`] = `
"Unexpected token * (1:14)
> 1 | import defer * as ns from "x" with { attr: "val" };
    |              ^
  2 |"
`;

exports[`import-defer-attributes-declaration.js [meriyah] format 1`] = `
"Import source must be string (1:14)
> 1 | import defer * as ns from "x" with { attr: "val" };
    |              ^
  2 |"
`;

exports[`import-defer-attributes-declaration.js format 1`] = `
====================================options=====================================
parsers: ["babel"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import defer * as ns from "x" with { attr: "val" };

=====================================output=====================================
import defer * as ns from "x" with { attr: "val" };

================================================================================
`;

exports[`no-default.js [__babel_estree] format 1`] = `
"Only \`import defer * as x from "./module"\` is valid. (1:14)
> 1 | import defer x from "x";
    |              ^
  2 |"
`;

exports[`no-default.js [acorn] format 1`] = `
"Unexpected token (1:14)
> 1 | import defer x from "x";
    |              ^
  2 |"
`;

exports[`no-default.js [babel] format 1`] = `
"Only \`import defer * as x from "./module"\` is valid. (1:14)
> 1 | import defer x from "x";
    |              ^
  2 |"
`;

exports[`no-default.js [espree] format 1`] = `
"Unexpected token x (1:14)
> 1 | import defer x from "x";
    |              ^
  2 |"
`;

exports[`no-default.js [meriyah] format 1`] = `
"Import source must be string (1:14)
> 1 | import defer x from "x";
    |              ^
  2 |"
`;

exports[`no-named.js [__babel_estree] format 1`] = `
"Only \`import defer * as x from "./module"\` is valid. (1:16)
> 1 | import defer { x } from "x";
    |                ^
  2 |"
`;

exports[`no-named.js [acorn] format 1`] = `
"Unexpected token (1:14)
> 1 | import defer { x } from "x";
    |              ^
  2 |"
`;

exports[`no-named.js [babel] format 1`] = `
"Only \`import defer * as x from "./module"\` is valid. (1:16)
> 1 | import defer { x } from "x";
    |                ^
  2 |"
`;

exports[`no-named.js [espree] format 1`] = `
"Unexpected token { (1:14)
> 1 | import defer { x } from "x";
    |              ^
  2 |"
`;

exports[`no-named.js [meriyah] format 1`] = `
"Import source must be string (1:14)
> 1 | import defer { x } from "x";
    |              ^
  2 |"
`;
